Selenium

brainmagmo opened the pull request.

brainmagmo committed:
Built POM base

brainmagmo committed:
..

brainmagmo committed:
Detailed 2/3 new tests for selenium

brainmagmo committed:
Implemented last test

brainmagmo committed:
Tests are now human readable through POM

dentaylor commented on ampeg.testing/src/test/java/AmpegTest.java:
```diff
@@ -0,0 +1,52 @@
+		System.setProperty("webdriver.chrome.driver", chromeDpath);
+
+		this.driver = new ChromeDriver();
+//		this.driver.manage()
```
Remove comments

brainmagmo replied:
OK

dentaylor commented on ampeg.testing/src/test/java/AmpegTest.java:
```diff
@@ -0,0 +1,52 @@
+		this.driver.manage()
+			.timeouts()
+			.implicitlyWait(10, TimeUnit.SECONDS)
+			;
```
Terminators do not need to be on their own lines.

brainmagmo replied:
OK

dentaylor commented on ampeg.testing/src/test/java/BlackTeePage.java:
```diff
@@ -0,0 +1,33 @@
+		.click()
+		;
+		try {
+			Thread.sleep(5000);
```
No thread sleeps allowed.

brainmagmo replied:
will work on this

dentaylor commented on ampeg.testing/src/test/java/BlackTeePage.java:
```diff
@@ -0,0 +1,33 @@
+		find(By.cssSelector("button#product-addtocart-button"))
+		.click()
+		;
+		try {
```
Do not use exceptions to test for conditions. Remove the try/catch

brainmagmo replied:
will be resolved with thread.sleep

dentaylor commented on ampeg.testing/src/test/java/PageObject.java:
```diff
@@ -0,0 +1,21 @@
+		return this.driver.getCurrentUrl();
+	}
+	
+	protected WebElement find(By by) {
```
Not sure if this makes things easier. You could replace this approach with PageFactory.InitElements() and it would be cleaner and more descriptive.

brainmagmo replied:
will work on this

dentaylor commented on ampeg.testing/src/test/java/HeritageManualsPage.java:
```diff
@@ -0,0 +1,17 @@
+	}
+
+	public PDFpage openFirstPDF() {
+		find(By.xpath("//table/tbody/tr[1]/td[2]"))
```
What if the table structure changes? How will you know if this is still the first PDF?

brainmagmo replied:
The top left cell should not be empty otherwise the table is poorly formated.
I changed the xpath to be more specific

dentaylor commented on ampeg.testing/src/test/java/ProductLinkTest.java:
```diff
@@ -0,0 +1,73 @@
+	@Test
+	public void downloadUserManual() {
+		
+		String expectedURL = "https://ampeg.com/data/6/0a020a40dc385e0b9cc618bd0/application/pdf/Heritage%2050th%20Anniversary%20SVT%20Quick%20Start%20Guide%20-%20English%20.pdf";
```
0a020a40dc385e0b9cc618bd0
This appears to be a dynamic value. If so, how often will it change?

brainmagmo replied:
changed test to only look at file name

brainmagmo committed:
Created Driver Factory Pattern Base

brainmagmo committed:
Finished basic driver manager factory.

brainmagmo committed:
Collapsed floating semicolons onto final line of multi-line "lines" of
code.

brainmagmo committed:
made test more robust

brainmagmo committed:
Replaced thread.sleep with WebDriverWait(..).until(..)

brainmagmo committed:
Changes addItemToCart to be in line stylistically with other tests

brainmagmo committed:
Implements @FindBy decorator pattern

brainmagmo committed:
Completes deprecation of find()

Previous commit also deprecated ensureArtistOnProductDetailsPage test
because the pages used are no longer on the public site

brainmagmo committed:
Merge pull request #15 from brainmagmo/driver-factory

Driver factory
